connection-loader/src/main/java/com/terracotta/connection/api/TerracottaConnectionService.java

```diff
@@ -75,11 +75,22 @@ public Connection connect(URI uri, Properties properties) throws ConnectionExcep
     TerracottaClientConfigParams clientConfig = new TerracottaClientConfigParams();
     String[] hosts = uri.getSchemeSpecificPart().split(",");
     for(String host : hosts) {
+
+      String[] tokens = host.split(":");
+      if(tokens.length >= 2) {
```

Comment:
Probably should throw an exception if greater than 2.  I wonder if this will mess up with ipv6

Reply:
okay, will make changes to throw an exception. this change will not work with ipv6